internal/controller/controllers/agent_controller.go

```diff
@@ -184,6 +184,17 @@ func (r *AgentReconciler) Reconcile(origCtx context.Context, req ctrl.Request) (
 				return r.updateStatus(ctx, log, agent, nil, nil, err2, !IsUserError(err2))
 			}
 			return r.updateStatus(ctx, log, agent, &host.Host, cluster.ID, nil, true)
+		} else if h.ClusterID != cluster.ID {
+			log.Debugf("ClusterDeploymentName is changed in Agent %s. unbind first", agent.Name)
+
+			host, err2 := r.Installer.UnbindHostInternal(ctx, installer.UnbindHostParams{
```

Comment:
looks like you have this logic twice https://github.com/openshift/assisted-service/blob/286b9294e4d8fe3e8a8cc3d2ac46b2831528d98e/internal/controller/controllers/agent_controller.go#L138-L145, consider adding a helper function.

Reply:
Done